src/map/clif.cpp

```diff
@@ -1749,6 +1749,36 @@ int clif_spawn( struct block_list *bl, bool walking ){
 	return 0;
 }
 
+/// Notifies client of a change in an elemental's status parameter.
+/// 0x7db <type>.W <value>.L (ZC_HO_PAR_CHANGE)
+/// 0xba5 <type>.W <value>.Q (ZC_HO_PAR_CHANGE2)
+void clif_homunculus_updatestatus(map_session_data& sd, _sp type) {
+#if PACKETVER >= 20090610
+	if( !hom_is_active(sd.hd) )
+		return;
+
+	PACKET_ZC_HO_PAR_CHANGE p = {};
+
+	p.packetType = HEADER_ZC_HO_PAR_CHANGE;
+	p.type = static_cast<decltype(p.type)>(type);
+	status_data* status = &sd.hd->battle_status;
+
+	switch (type) {
+	case SP_BASEEXP:
+		p.value = static_cast<decltype(p.value)>(sd.hd->homunculus.exp);
```

Comment:
Here i see a problem.
Currently we only have a check for player exp, which were changed 2017 from int32 to int64.
For homunculus we do not have something like that, but it seems to be supported since 2021 as well (see also clif_hominfo).